ipamapi/contract.go

```diff
@@ -42,59 +51,22 @@ type AddressSpace string
 // Common key is a addressspace
 type Config interface {
 	// AddSubnet adds a subnet to the specified address space
-	AddSubnet(AddressSpace, *SubnetInfo) error
+	AddSubnet(string, *net.IPNet) error
 	// RemoveSubnet removes a subnet from the specified address space
-	RemoveSubnet(AddressSpace, *net.IPNet) error
-	// AddVendorInfo adds Vendor specific data
-	AddVendorInfo([]byte) error
-}
-
-// SubnetInfo contains the information subnet hosts need in order to communicate
-type SubnetInfo struct {
-	Subnet     *net.IPNet
-	Gateway    net.IP
-	OpaqueData []byte // Vendor specific
+	RemoveSubnet(string, *net.IPNet) error
 }
 
 /*************************
  * IPAM Service Interface
  *************************/
 
-// IPAM defines the interface that needs to be implemented by IPAM service plugin
+// Allocator defines the interface the IPAM service plugins must implement
 // Common key is a unique address space identifier
-type IPAM interface {
+type Allocator interface {
 	// Request address from the specified address space
-	Request(AddressSpace, *AddressRequest) (*AddressResponse, error)
+	Request(string, *net.IPNet, net.IP) (net.IP, error)
```

Comment:
I will change the description to make it clearer.
Personally I prefer the raw string type to a typedef of a string. I've seen how it can become annoying to typecast strings to the new type.